### PR TITLE
Test submission 

### PR DESCRIPTION
**Pull request by vrackovic**
- After applicant login, user is redirected to newly generated test
- Test creates form fields and binds them to `NewTestForm`
- After test is done, it is submitted to `/api/tests/submit`
- Payload normalized to match endpoints requirements

Resolves issues #223

### CODE REVIEW

**Comment by iscekic** [comment]
Please resolve conflicts and I'll review. 😄

File: web/src/main/frontend/src/base/rest/testApi.js [hunk 1]
```diff
@@ -9,7 +9,16 @@ import {getLink} from '../getLink';
...
+      request
+        .get(getLink(`/tests/${testId}`))
+        .set(testApi.token)
+        .then(res => res.body)
+    );
+  }
+
+  submitTest(testSolution) {
+    return withLoader(
+      request.get(getLink('/tests/submit', testSolution)).set(testApi.token)
```
Comment (iscekic) [inline web/src/main/frontend/src/base/rest/testApi.js]: We don't need the body of this request, right? 😄

File: web/src/main/frontend/src/base/util.js [hunk 2]
```diff
@@ -53,3 +54,35 @@ export const questionFormNormalizer = questions => {
     category: value
   };
 };
+
+export const codingChallengeQuestionFormating = answer =>
+  answer && answer.getValue('\n');
```
Comment (iscekic) [inline web/src/main/frontend/src/base/util.js]: What does this do exactly?

File: web/src/main/frontend/src/base/util.js [hunk 3]
```diff
@@ -53,3 +54,35 @@ export const questionFormNormalizer = questions => {
...
+
+export const checkQuestionType = (acc, question) => {
+  let normalizedForm = {};
+  const {questionId, questionType, answer} = question;
+  switch (question.questionType) {
```
Comment (iscekic) [inline web/src/main/frontend/src/base/util.js]: `switch(questionType)`

File: web/src/main/frontend/src/components/utility/getComponentByQuestionType.js [hunk 4]
```diff
@@ -4,75 +4,88 @@ import TextAreaQuestion from '../TextAreaQuestion';
...
-          {reviewMode && (
-            <Points
-              id={props.id}
-              points={props.availablePoints}
-              valuatedScore
+@inject('newTestForm')
+@observer
+class GetComponentByQuestionType extends React.Component {
+  render() {
+    const answers = this.props.newTestForm.$('answers').add();
```
Comment (iscekic) [inline web/src/main/frontend/src/components/utility/getComponentByQuestionType.js]: Are you sure this should be part of `render`? (Same question for `MultipleChoiceQuestion`)

File: web/src/main/frontend/src/components/utility/getComponentByQuestionType.js [hunk 5]
```diff
@@ -4,75 +4,88 @@ import TextAreaQuestion from '../TextAreaQuestion';
...
+              question={this.props.question.text}
+              options={this.props.question.answers}
+              solution={this.props.question.solutionID}
+              reviewMode={this.props.question.reviewMode}
+              answer={answers}
+              isCorrect={
+                this.props.question.solutionID &&
+                this.props.question.options.find(
+                  opt => opt.id === this.props.question.solutionID
+                ).isCorrect
```
Comment (iscekic) [inline web/src/main/frontend/src/components/utility/getComponentByQuestionType.js]: Potential error if `find` finds nothing. Double-check.

File: web/src/main/frontend/src/components/utility/getComponentByQuestionType.js [hunk 6]
```diff
@@ -4,75 +4,88 @@ import TextAreaQuestion from '../TextAreaQuestion';
...
-        </div>
-      );
-    default:
-      return null;
+            {this.props.reviewMode && (
+              <Points
+                id={this.props.id}
+                points={this.props.availablePoints}
+                fixedScore
+                // TODO: Uncomment and fix values after `test submit` has been implemented
```
Comment (iscekic) [inline web/src/main/frontend/src/components/utility/getComponentByQuestionType.js]: Should this `TODO` be implemented now?

File: web/src/main/frontend/src/base/util.js (hunk 2 above)
Comment (vrackovic) [reply web/src/main/frontend/src/base/util.js]: Its from `CodeMirror`. Without it, value entered in multiple lines of `CodeMirror` will be merged into one line.

File: web/src/main/frontend/src/components/utility/getComponentByQuestionType.js (hunk 6 above)
Comment (vrackovic) [reply web/src/main/frontend/src/components/utility/getComponentByQuestionType.js]: This PR is about submitting tests, and `TODO` needs to be implemented for `testReview`. That should be seperate PR

File: web/src/main/frontend/src/base/rest/testApi.js (hunk 1 above)
Comment (vrackovic) [reply web/src/main/frontend/src/base/rest/testApi.js]: Not sure. After submitting test, user should be logged out and redirected to home page.
Maybe some data is needed to be sure that test is successfully submitted, and after that do redirection and logout?

File: web/src/main/frontend/src/components/utility/getComponentByQuestionType.js (hunk 4 above)
Comment (vrackovic) [reply web/src/main/frontend/src/components/utility/getComponentByQuestionType.js]: Will be moved outside of `render`

File: web/src/main/frontend/src/components/utility/getComponentByQuestionType.js (hunk 5 above)
Comment (vrackovic) [reply web/src/main/frontend/src/components/utility/getComponentByQuestionType.js]: Same as next comment, it is part of `reviewTest` component and will be fixed with PR made only for that purpose

**Pull request merged**